rt/rs/client/src/main/java/org/apache/cxf/jaxrs/client/JAXRSClientFactoryBean.java

```diff
@@ -403,13 +412,45 @@ protected void initClient(AbstractClient client, Endpoint ep, boolean addHeaders
             });
         }
     }
+    
+    protected <C extends Configurable<C>> Configurable<?> getConfigurableFor(C context) {
```

Comment:
Just thinking that this method might be better off named `newConfigurableFor` to avoid confusion.  My first thought was that this method might be used to cache the configurable based on the passed-in context - like a `Map<C, Configurable<C>>`.  No worries if you'd prefer to keep it as-is though.

Reply:
Thanks a lot for looking, @andymc12 ! I was also initially though of "create" naming but in case of MP, it turned out to be not exactly the case: the existing instance of the configurable is used for any context. In this regards, `getConfigurableFor` is a bit more honest name with respect to semantics, but `getOrCreateConfigurableFor` would also make sense. What do you think?

Reply:
Oh - I totally missed that.  I think either way (`getConfigurableFor` or `getOrCreateConfigurableFor`) is fine.  Thanks for the clarification.